todo_app/deploy.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+echo $DOCKER_PASSWORD | docker login --username $DOCKER_USER --password-stdin
+docker build --target production --cache-from kevineatondevops/todo-app:latest --tag kevineatondevops/todo-app:latest --tag kevineatondevops/todo-app:$TRAVIS_COMMIT --tag registry.heroku.com/kedevopstodoapp/web .
+docker push kevineatondevops/todo-app:latest
+docker login --username=_ --password=$(heroku auth:token) registry.heroku.com
```

Comment:
Here we should pass the heroku auth token in a similar fashion to the docker password above (i.e. using `--password-stdin`) in order to avoid potentially leaking the password in shell logs. [See the docker docs for more info](https://docs.docker.com/engine/reference/commandline/login/) 

Also, worth noting that you could use your `$HEROKU_API_KEY` directly as a password here rather than using it to silently log into the heroku cli in order to generate a new token, but no strong advantage.